### PR TITLE
feat: add domain line to resolv.conf

### LINKED ISSUE #1626: use domainname for target suffix
If we have a domainname on a node and we are given a target (node) which is an unresolvable name, attempt to append the domainname before giving up.

This is hardly a high-priority request, but merely a convenience to have Talos's apid proxy act a little more like standard domain resolution, by attempting to resolve the hostname by appending the domainname (exactly as a `domain` line in `resolv.conf` would do).

In fact, it would probably be sufficient to add a `domain` line to the local `resolv.conf`, and it is likely Go's `net.Dial` will just do the right thing.

### PR DESCRIPTION
**Pull request by Ulexus**
If we have a domainname, add it to resolv.conf.

Fixes #1626

Signed-off-by: Seán C McCord <ulexus@gmail.com>

### CODE REVIEW

**Comment by bradbeam** [comment]
Would you mind using the search key instead of domain name?

File: internal/app/networkd/pkg/networkd/misc.go [hunk 1]
```diff
@@ -12,6 +12,9 @@ import (
 	"net"
 	"strings"
 	"text/template"
+
+	"github.com/pkg/errors"
```
Comment (andrewrynhard) [inline internal/app/networkd/pkg/networkd/misc.go]: We moved to standard library wrapping.

File: internal/app/networkd/pkg/networkd/misc.go [hunk 2]
```diff
@@ -49,6 +52,14 @@ func writeResolvConf(resolvers []string) (err error) {
 		}
 	}
 
+	if domain, err := talosnet.DomainName(); err == nil {
+		if domain != "" {
+			if _, err = resolvconf.WriteString(fmt.Sprintf("domain %s\n", domain)); err != nil {
+				return errors.Wrap(err, "failed to add domain line to resolvconf")
```
Comment (andrewrynhard) [inline internal/app/networkd/pkg/networkd/misc.go]: Can we make this `fmt.Errorf("failed to add domain line to resolvconf: %w", err)`

**Comment by Ulexus** [comment]
Well, we could certainly use the search key, but this _is_ what the domain key is for.  Is there a reason for that preference?

File: internal/app/networkd/pkg/networkd/misc.go (hunk 1 above)
Comment (Ulexus) [reply internal/app/networkd/pkg/networkd/misc.go]: Ah, yeah.  Wishing he would update his library to use stdlib with the same API.  Oh well.

**Comment by andrewrynhard** [comment]
I'm kinda impartial but my 2 cents: Since `domain` essentially gives us the same thing as `search`, and it aligns with the kernel domain name, I would lean towards using `domain`.

**Comment by bradbeam** [comment]
I was under the impression domain was mutually exclusive with search, but according to this man page domain is obsolete 
http://man7.org/linux/man-pages/man5/resolv.conf.5.html

**Comment by andrewrynhard** [comment]
> I was under the impression domain was mutually exclusive with search, but according to this man page domain is obsolete
> http://man7.org/linux/man-pages/man5/resolv.conf.5.html

Good find. That changes my mind :).

**Comment by Ulexus** [comment]
Likewise...  My local man page doesn't have the same wording, so I guess _it_ is obsolete.

**Comment by andrewrynhard** [comment]
@Ulexus Wanna move the PR out of draft. @bradbeam You LGTM in this?

**Comment by Ulexus** [comment]
Yes, just testing first

**Comment by Ulexus** [comment]
Yep, works perfectly to fix #1626

**Pull request merged**